src/nurse/views.py

```diff
@@ -57,6 +58,13 @@ class NurseViewSet(viewsets.ModelViewSet):
     serializer_class = NurseSerializer
 
 
+class UserOneSignalProfileViewSet(
+    mixins.CreateModelMixin, viewsets.ReadOnlyModelViewSet
+):
+    queryset = UserOneSignalProfile.objects.all()
+    serializer_class = UserOneSignalProfileSerializer
```

Comment:
Good start, on the view however we want to make sure only the logged user has access to his one signal profile.
Look how we did it for `PatientViewSet`
Basically with `get_queryset()` you make sure that only the UserOneSignalProfile associated to the user gets returned. And with `create()` you make sure that the user.id gets retrieved from the request object rather than passed as payload.

Reply:
Addressed thanks